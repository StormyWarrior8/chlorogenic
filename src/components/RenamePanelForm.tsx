import * as React from "react"
import styled from "../appearance/styled"
import Button from "./parts/Button"
import Flexbox, { VFlexbox } from "./parts/Flexbox"

const Title = styled.h3`
  margin: 0;
  margin-bottom: 0.4em;
  padding: 0;
`

export interface RenamePanelFormProps {
  defaultName?: string
  onClickOk?(name: string): void
  onClickCancel?(): void
  onClickDelete?(): void
}

class RenamePanelForm extends React.PureComponent<RenamePanelFormProps> {
  private inputRef = React.createRef<HTMLInputElement>()
  public render() {
    const { defaultName, onClickCancel, onClickOk, onClickDelete } = this.props
    return (
      <VFlexbox>
        <Title>Edit name</Title>
        <input
          autoFocus={true}
          defaultValue={defaultName}
          ref={this.inputRef}
        />
        <Flexbox>
          <Button onClick={onClickDelete} negative={true}>
            Delete
          </Button>
          <Button onClick={onClickCancel}>Cancel</Button>
          <Button
            onClick={() =>
              this.inputRef &&
              onClickOk &&
              onClickOk(this.inputRef.current!.value)
            }
            positive={true}
          >
            Change name
          </Button>
        </Flexbox>
      </VFlexbox>
    )
  }
}

export default RenamePanelForm

const ModalStyle = styled(Flexbox)`
  margin: 1em;
`

export const RenamePanelFormDialog = (props: RenamePanelFormProps) => (
  <ModalStyle>
    <RenamePanelForm {...props} />
  </ModalStyle>
)
