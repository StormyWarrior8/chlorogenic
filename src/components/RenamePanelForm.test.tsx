import * as React from "react"

import { render } from "react-testing-library"
import RenamePanelForm from "./RenamePanelForm"

it("snapshot", () => {
  {
    const { container } = render(<RenamePanelForm />)
    expect(container.firstChild).toMatchSnapshot()
  }

  {
    const { container } = render(
      <RenamePanelForm
        defaultName="(◍•ᴗ•◍)"
        onClickCancel={jest.fn()}
        onClickDelete={jest.fn()}
        onClickOk={jest.fn()}
      />,
    )
    expect(container.firstChild).toMatchSnapshot()
  }
})
