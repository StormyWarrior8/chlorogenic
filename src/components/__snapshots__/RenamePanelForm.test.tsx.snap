// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="sc-bxivhb glCqXO"
>
  <h3
    class="sc-ifAKCX lfULYF"
  >
    Edit name
  </h3>
  <input
    value=""
  />
  <div
    class="sc-htpNat cixthU"
  >
    <button
      class="sc-bdVaJa dWNBmp"
    >
      Delete
    </button>
    <button
      class="sc-bdVaJa dnqbXl"
    >
      Cancel
    </button>
    <button
      class="sc-bdVaJa bNhLRi"
    >
      Change name
    </button>
  </div>
</div>
`;

exports[`snapshot 2`] = `
<div
  class="sc-bxivhb glCqXO"
>
  <h3
    class="sc-ifAKCX lfULYF"
  >
    Edit name
  </h3>
  <input
    value="(◍•ᴗ•◍)"
  />
  <div
    class="sc-htpNat cixthU"
  >
    <button
      class="sc-bdVaJa dWNBmp"
    >
      Delete
    </button>
    <button
      class="sc-bdVaJa dnqbXl"
    >
      Cancel
    </button>
    <button
      class="sc-bdVaJa bNhLRi"
    >
      Change name
    </button>
  </div>
</div>
`;
